src/withdraw-multi-token.circom

```diff
@@ -0,0 +1,192 @@
+include "../node_modules/circomlib/circuits/comparators.circom";
+include "../node_modules/circomlib/circuits/poseidon.circom";
+include "../node_modules/circomlib/circuits/smt/smtverifier.circom";
+include "../node_modules/circomlib/circuits/sha256/sha256.circom";
+
+include "./lib/hash-state.circom"
+
+/**
+ * Verify withdrawal by proving that a leaf exist on the exit tree
+ * @param nLevels - merkle tree depth
+ * @input rootState - {Field} - state tree root
+ * @input ethAddr - {Uint160} - ethereum address
+ * @input tokenIDs - {Array(Uint32)} - token identifiers
+ * @input nonces - {Array(Uint40)} - nonces
+ * @input balances - {Array(Uint192)} - balances
+ * @input idxs - {Array(Uint48)} - merkle tree indexes
+ * @input exitBalances - {Array(Uint192)} - exitBalances
+ * @input accumulatedHashes - {Array(Field)} - accumulatedHashes
+ * @input signs - {Array(Bool)} - babyjubjub signs
+ * @input ays - {Array(Field)} babyjubjub y coordinate
+ * @input siblingsStates[nLevels + 1] - {Array(Array(Field))} - siblings merkle proof
+ * @output hashGlobalInputs - {Field} - hash of all pretended input signals
+ */
+template WithdrawMultiToken(nLevels, nTokens) {
+    // Unique public signal
+    signal output hashGlobalInputs;
+
+    // private inputs
+    signal private input rootState;
+	signal private input ethAddr;
+    signal private input tokenIDs[nTokens];
+    signal private input nonces[nTokens];
+    signal private input balances[nTokens];
+    signal private input idxs[nTokens];
+    signal private input exitBalances[nTokens];
+    signal private input accumulatedHashes[nTokens];
+    signal private input signs[nTokens];
+    signal private input ays[nTokens];
+    signal private input siblingsStates[nTokens][nLevels + 1];
+
+    var i;
+
+    // compute account state hash
```

Comment:
I would move this comment just right before the initialization of `HashState` component